doc/HOWTO/keys.txt

```diff
@@ -68,5 +68,42 @@ you don't want your key to be protected by a password, remove the flag
     type in the password every time the server needs to access
     the key.
 
+
+4. To generate an EC key
+
+An EC key can be used both for key agreement (ECDH) and signing (ECDSA).
+
+Generating a key for ECC is similar to generating a DSA key. These are
+two-step processes. First, you have to generate EC parameters from which
+the key will be generated:
+
+  openssl ecparam -name prime256v1 -out prime256v1.pem
+
+The prime256v1 (a.k.a NIST P-256), which stands for 'X9.62/SECG curve over
+a 256 bit prime field', is the name of an elliptic curve which generates the
+parameters. You can use the following command to list all supported curves
+in OpenSSL:
+
+  openssl ecparam -list_curves
+
+When that is done, you can generate a key using the created parameters (several
+keys can be produced from the same parameters):
+
+  openssl genpkey -des3 -paramfile prime256v1.pem -out private.key
+
+With this variant, you will be prompted for a password to protect your key.
+If you don't want your key to be protected by a password, remove the flag
+'-des3' from the command line above.
+
+    NOTE: if you intend to use the key together with a server
+    certificate, it may be good to avoid protecting it with a
+    password, since that would mean someone would have to type
+    in the password every time the server needs to access the
+    key.
+
+You can also directly generate the key in one step:
+
+  openssl ecparam -genkey -name prime256v1 -out private.key
+
```

Comment:
You can also generate the key in one step with `genpkey`:
```
openssl genpkey -algorithm EC -pkeyopt ec_paramgen_curve:P-256
```

We might want to note that X25519 (and Ed25519 in the master branch) are treated as distinct algorithms and **not** usable with `ecparam` et al.